src/transformers/convert_slow_tokenizer.py

```diff
@@ -596,7 +597,11 @@ def tokenizer(self, proto):
         # user defined symbols are not
         # both user and control tokens are AddedTokens
         # Add user defined symbols (type == 4) from sentencepiece (https://github.com/google/sentencepiece/blob/6225e08edb2577757163b3f5dbba4c0b670ef445/src/sentencepiece_model.proto#L299C29-L299C33)
-        spm_added_tokens = [(id, p.piece, p.type == 3) for id, p in enumerate(proto.pieces) if p.type in [3, 4]]
+        spm_added_tokens = [
+            (id, p.piece, p.type == 3 or p.piece in self.special_tokens)
```

Comment:
Was the `self.special_tokens` condition necessary for some tokenizers? I think that `<start_of_turn>`, `<end_of_turn>` were already added in Gemma, but not as special because of the missing property below.

Also, maybe update the comment below this block if you think it can be clearer.

Reply:
Other tokenizers won't be affected since `self.special_tokens` defaults to `{}`. This way just makes it easier to add support for other tokenizers in future.

Indeed, the Gemma tokenizer (instruct) needs to add these as special tokens. It isn't needed for the base model, but shouldn't affect anything since those tokens are only used for instructions